src/app/pages/service/photos/photos.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http'
+import { environment } from '../../../environments/environment'
+
+@Injectable({
+    providedIn: 'root'
+})
+
+export class PhotosService {
+
+    constructor(private http: HttpClient) { }
+
+    getPhotos(): Promise<Photo[]> {
+        return this.http.get(`https://${environment.facebookphotosUrl}`, {
+        params: { token: environment.facebookphotosToken }
+    }).toPromise()
+    .then((response: any) =>
+    //is this the correct way to make an api call?
+        FB.api(
+        "/{photo-id}",
+        function (response) {
+            if (response && !response.error) {
+                link: link,
+                name: name,
+            }
```

Comment:
Sorry, I don't quite understand this part.
Maybe it's a dumb question, but why did you have an object directly after if statement?
Does this return or something?

Reply:
I think `if (object)` checks for the definition of an object. If an object is defined, it is somehow `truthy`. Likewise, if an object is undefined, it will be `falsy`. It is the same as if `(response != undefined)`.

Reply:
Yea this one I understand.
But inside the if statement I mean
```javascript
{
  link: link,
  name: name
}
```
Where does the value return to?
Is this like a state change in Angular?